### DISCUSSION TITLE
Image uploaded to bucket is corrupted

### DISCUSSION THREAD

**Discussion by pabichw** [supabase/supabase#7222] [Questions] [answered]
I try to upload a jpeg image to bucket. It works fine when I don't specify `content-type`. When I set it it `"content-type": "image/jpg"` upload is successful, but the image is corrupted (downloaded image cannot be opened and under url there is only small white square).

I'm using the following code: 
```
url: str = SUPABASE_URL
key: str = SUPABASE_KEY
supabase: Client = create_client(url, key)

filename = os.fsdecode(file)
filepath = os.path.join(path, filename)

 supabase.storage().from_('my-bucket-xxx').upload(
    path='public/%s' % (filename), 
    file=filepath, 
    file_options={ "content-type": "image/jpg" }
)
```

Thanks for any help

**Comment by GaryAustin1** [comment]
You might try image/jpeg for the mime type.

**Reply by pabichw** [reply]
thanks for suggestion, I've tried it but image is still corrupted

**Comment by pabichw** [comment, marked as answer]
I ended up rewriting script to Node.js and same logic works perfectly fine. I highly discourage from using python client...

**Comment by shravan-d** [comment]
I am currently using the same code in Node for a react native application, and I am only getting a corrupted image. The uploaded image has a size of a few hundred kb so I know it is getting uploaded. I've checked the base64data as well, and that seems to be alright. Here my code:

> var base64Data = await ImgToBase64.getBase64String(`file://${path}`);
>     const { data, error } = await supabase
>       .storage
>       .from('avatars')
>       .upload('public/avatar11.jpg', decode(base64Data), {
>         contentType: 'image/jpeg',
>         upsert: true,
>       })
>     console.log(data, error)

Any ideas for a fix?

**Reply by gk258007** [reply]
Did you find any fix for this?

**Reply by pierroo** [reply]
Joining the party too;
constantly getting a broken image in supabase when the base64 and decoded file are indeed correct;

Edit: actually I just found a fix, instead of using decode from "base64-arraybuffer" like in supabase upload example, I have used the function from here: https://github.com/danguer/blog-examples/blob/master/js/base64-binary.js
So apparently the issue was from the decode function...

**Reply by Giuva-Sergi** [reply]
How did you fix it? Could you show me the code because I don't know which function to use in that file

**Reply by pierroo** [reply]
@Giuva-Sergi I use this:

`const res = Base64Binary.decodeArrayBuffer(base64Str)`
The base64Str is the file I want to upload from my app
That res is the one I upload;
And that Base64Binary function is this:

```
const Base64Binary = {
  _keyStr : "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=",
  
  /* will return a  Uint8Array type */
  decodeArrayBuffer: function(input) {
    var bytes = (input.length/4) * 3;
    var ab = new ArrayBuffer(bytes);
    this.decode(input, ab);
    
    return ab;
  },

  removePaddingChars: function(input){
    var lkey = this._keyStr.indexOf(input.charAt(input.length - 1));
    if(lkey == 64){
      return input.substring(0,input.length - 1);
    }
    return input;
  },

  decode: function (input, arrayBuffer) {
    //get last chars to see if are valid
    input = this.removePaddingChars(input);
    input = this.removePaddingChars(input);

    var bytes = parseInt((input.length / 4) * 3, 10);
    
    var uarray;
    var chr1, chr2, chr3;
    var enc1, enc2, enc3, enc4;
    var i = 0;
    var j = 0;
    
    if (arrayBuffer)
      uarray = new Uint8Array(arrayBuffer);
    else
      uarray = new Uint8Array(bytes);
    
    input = input.replace(/[^A-Za-z0-9\+\/\=]/g, "");
    
    for (i=0; i<bytes; i+=3) {	
      //get the 3 octects in 4 ascii chars
      enc1 = this._keyStr.indexOf(input.charAt(j++));
      enc2 = this._keyStr.indexOf(input.charAt(j++));
      enc3 = this._keyStr.indexOf(input.charAt(j++));
      enc4 = this._keyStr.indexOf(input.charAt(j++));
  
      chr1 = (enc1 << 2) | (enc2 >> 4);
      chr2 = ((enc2 & 15) << 4) | (enc3 >> 2);
      chr3 = ((enc3 & 3) << 6) | enc4;
  
      uarray[i] = chr1;			
      if (enc3 != 64) uarray[i+1] = chr2;
      if (enc4 != 64) uarray[i+2] = chr3;
    }
  
    return uarray;	
  }
}
```

**Comment by Bryson14** [comment]
Having the same issue with the python client. Not sure what the issue is.
